Game/Resources/final_event/MonsterScript.as

```diff
@@ -0,0 +1,83 @@
+class MonsterScript {
+    Hub @hub;
+    Entity @self;
+	int phase = 0;
+	float fallspeed = 0.0f;
+	float waitTimer = 0.0f;
+	float eatingTimer = 0.0f;
+	Entity @cart;
+
+    MonsterScript(Entity @entity){
+        @hub = Managers();
+        @self = @entity;
+
+        // Remove this if updates are not desired.
+        RegisterUpdate();
+    }
+
+    // Called by the engine for each frame.
+    void Update(float deltaTime) {
+		switch (phase) {
+			case 0: // Waiting
+				break;
+			case 1: { // Jump out
+				fallspeed += 9.82f * deltaTime;
+				self.position.y -= fallspeed * deltaTime;
+				if (self.position.y <= 0.0f) {
+					self.position.y = 0.0f;
+					phase = 2;
+				}
+				break;
+			}
+			case 2: { // Landing
+				waitTimer += deltaTime;
+				if (waitTimer >= 1.0f) {
+					phase = 3;
+				}
+				break;
+			}
+			case 3: { // Approach player
+				self.position.x -= 6.0f * deltaTime;
+				break;
+			}
+			case 5: { // Eating
```

Comment:
case 4?

Reply:
The fourth phase is when the monster waits for the player to enter its warm embrace (transitioned into via `StopCharging`). It has no special logic which is why there is no case for it.